pandas/_libs/tslibs/timestamps.pyi

```diff
@@ -56,7 +54,7 @@ class Timestamp(datetime):
         tzinfo: _tzinfo | None = ...,
         *,
         fold: int | None = ...,
-    ) -> _DatetimeT | NaTType: ...
+    ) -> _DatetimeT: ...
```

Comment:
can you add a comment (maybe ref this issue) about the reasoning here.

Reply:
done in next commit